### DISCUSSION TITLE
Failed to load Pgtcl error while running librarycheck :

### DISCUSSION THREAD

**Discussion by noopur100** [TPC-Council/HammerDB#429] [Q&A] [answered]
I am able to buildschema without error but when I run script to run test load then it gives below error.
mysql files are stored at location  '/opt/mysql-8.0.18-linux-glibc2.12-x86_64/lib'

Vuser 1:RUNNING
Transaction Counter Error: failed to load library couldn't load file "/hammerdb/HammerDB-4.5/lib/pgtcl2.1.1/libpgtcl2.1.1.so": libpq.so.5: cannot open shared object file: No such file or directory
Error in Virtual User 1: Failed to load Pgtcl - couldn't load file "/hammerdb/HammerDB-4.5/lib/pgtcl2.1.1/libpgtcl2.1.1.so": libpq.so.5: cannot open shared object file: No such file or directory
Vuser 1:FINISHED FAILED
Vuser 2:RUNNING
Error in Virtual User 2: Failed to load Pgtcl - couldn't load file "/hammerdb/HammerDB-4.5/lib/pgtcl2.1.1/libpgtcl2.1.1.so": libpq.so.5: cannot open shared object file: No such file or directory
Vuser 2:FINISHED FAILED

**Comment by sm-shaw** [comment, marked as answer]
First thing to verify is whether you are testing MySQL or PostgreSQL? You say mysql files are stored at location '/opt/mysql-8.0.18-linux-glibc2.12-x86_64/lib' but the error is for PostgreSQL?

So the error is libpq.so.5: cannot open shared object file: No such file or directory. libpq is the name of the PostgreSQL library that we need. To resolve this install the PostgreSQL client (also included with the server) and add the lib directory to the LD_LIBRARY_PATH environment variable. There is information on how to do this in the documentation here https://hammerdb.com/docs/ch01s10.html.

**Reply by noopur100** [reply]
yes, in human error I had forgot to update the postgres script for mysql thats why pg error was there and issue was resolved after setting LD_Library_path to location - opt/mysql-8.0.18-linux-glibc2.12-x86_64/lib
